Source/Pages/Gallery/YPLibraryVC.swift

```diff
@@ -374,7 +374,7 @@ public class YPLibraryVC: UIViewController, YPPermissionCheckable {
                                          callback: @escaping (_ videoURL: URL) -> Void) {
         if fitsVideoLengthLimits(asset: asset) == true {
             delegate?.libraryViewStartedLoading()
-            let normalizedCropRect = withCropRect ?? (Thread.isMainThread ? v.currentCropRect() : DispatchQueue.main.sync { v.currentCropRect() })
+            let normalizedCropRect = withCropRect ?? v.currentCropRect()
```

Comment:
I think you may have reverted this code incorrectly. It is now always missing a `DispatchQueue.main.sync` wrapper (unless that was also not supposed to be there), but `currentCropRect` will now for sure be called off of a main thread via line 418 for example, which I believe is being executed from a background thread from line 385, and that seems like something you wanted to avoid originally.

Reply:
Yeah you're right, I wasn't looking at the multiple selection case since we aren't using it in Fudos.

The crash happens when you call `DispatchQueue.main.sync` from the main thread, which isn't allowed. It looks weird here, but I think it makes more sense to just go back to the ternary operator than to try and track down everywhere that calls checkVideoLengthAndCrop and make sure its called on the main thread.